gwcs/tags/selectortags.py

```diff
@@ -0,0 +1,121 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, unicode_literals, print_function
+
+import numpy as np
+from numpy.testing import assert_array_equal
+from astropy.modeling import models
+from astropy.utils import OrderedDict
+from pyasdf import yamlutil
+from pyasdf.tags.transform.basic import TransformType
+from pyasdf.tags.core.ndarray import NDArrayType
+
+from ..selector import *
+
+
+__all__ = ['LabelMapperType', 'RegionsSelectorType']
+
+
+class LabelMapperType(TransformType):
+    name = "transform/label_mapper"
+    types = [LabelMapperArray, LabelMapperDict, LabelMapperRange]
+
+    @classmethod
+    def from_tree_transform(cls, node, ctx):
+        inputs_mapping = node.get('inputs_mapping', None)
+        if inputs_mapping is not None and not isinstance(inputs_mapping, models.Mapping):
+            raise TypeError("inputs_mapping must be an instance"
+                            "of astropy.modeling.models.Mapping.")
+        mapper = node['mapper']
+
+        if isinstance(mapper, NDArrayType):
+            if mapper.ndim != 2:
+                raise NotImplementedError(
+                    "GWCS currently only supports 2x2 masks ")
+            return LabelMapperArray(mapper, inputs_mapping)
+        else:
+            inputs = node.get('inputs', None)
+            labels = mapper.get('labels')
+            transforms = mapper.get('models')
+            dict_mapper = dict(zip(labels, transforms))
+            return LabelMapperDict(inputs, dict_mapper, inputs_mapping)
+
+    @classmethod
+    def to_tree_transform(cls, model, ctx):
+        node = OrderedDict()
+        if isinstance(model, LabelMapperArray):
+            node['mapper'] = model.mapper
+        if isinstance(model, (LabelMapperDict, LabelMapperRange)):
+            mapper = OrderedDict()
+            labels = list(model.mapper.keys())[:]
+
+            transforms = []
+            for k in labels:
+                transforms.append(model.mapper[k])
+            if not np.isscalar(labels[0]):
```

Comment:
If you are saying that a python tuple is converted to an yaml array, it's not what I see with pyasdf. Try writing this to a file ` m = models.Mapping((1,0))` (which is why I submitted  spacetelescope/pyasdf#169)